A2AClientAgent issue: jakarta.enterprise.inject.CreationException: Error creating synthetic bean [...]: java.lang.IllegalArgumentException: No agent method found in class: org.acme....

Trying to connect to a Remote A2A Server agent (listening on port 11001) using the A2A protocol. Created this A2A Client Agent below

the A2A Server Agent is protected by an OIDC scheme so in order to be able to connect to it the client agent needs to send a valid access token in the Authorization request header.

```
package org.acme;

import dev.langchain4j.agentic.declarative.A2AClientAgent;
import dev.langchain4j.service.UserMessage;

public interface BinanceAgenticA2aClientAgent {

    @A2AClientAgent(a2aServerUrl = "http://localhost:11001",
        outputKey = "aiMessage",
        description = "Binance Crypto Market Analizer")
    String analizeMarket(@UserMessage DeepChatMessages question);
}
```

run it like this

```
String aiMessageString = binanceAgenticA2aClientAgent.analizeMarket(mapper.readValue(question, DeepChatMessages.class));
        System.out.println("Received question: " + aiMessageString);
```

it throws this error

```
: jakarta.enterprise.inject.CreationException: Error creating synthetic bean [EtsuQd6QuW8aZwTzxOVk-1beTR0]: java.lang.IllegalArgumentException: No agent method found in class: org.acme.BinanceAgenticA2aClientAgent
	at org.acme.BinanceAgenticA2aClientAgent_EtsuQd6QuW8aZwTzxOVk-1beTR0_Synthetic_Bean.create(Unknown Source)
	at org.acme.BinanceAgenticA2aClientAgent_EtsuQd6QuW8aZwTzxOVk-1beTR0_Synthetic_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)
	at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.ch13(Unknown Source)
	at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.computeIfAbsent(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)
	at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:23)
	at org.acme.BinanceAgenticA2aClientAgent_EtsuQd6QuW8aZwTzxOVk-1beTR0_Synthetic_ClientProxy.arc$delegate(Unknown Source)
	at org.acme.BinanceAgenticA2aClientAgent_EtsuQd6QuW8aZwTzxOVk-1beTR0_Synthetic_ClientProxy.analizeMarket(Unknown Source)
	at org.acme.DeepChatAssistantResource.onTextMessage(DeepChatAssistantResource.java:65)
	at org.acme.DeepChatAssistantResource_WebSocketServerEndpoint.doOnTextMessage(Unknown Source)
	at io.quarkus.websockets.next.runtime.WebSocketEndpointBase$4.call(WebSocketEndpointBase.java:165)
	at io.quarkus.websockets.next.runtime.WebSocketEndpointBase$4.call(WebSocketEndpointBase.java:160)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$4(ContextImpl.java:192)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:270)
	at io.vertx.core.impl.ContextImpl$1.execute(ContextImpl.java:221)
	at io.vertx.core.impl.WorkerTask.run(WorkerTask.java:56)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.IllegalArgumentException: No agent method found in class: org.acme.BinanceAgenticA2aClientAgent
	at dev.langchain4j.agentic.internal.AgentUtil.validateAgentClass(AgentUtil.java:324)
	at dev.langchain4j.agentic.internal.AgentUtil.validateAgentClass(AgentUtil.java:305)
	at dev.langchain4j.agentic.internal.AgentUtil.validateAgentClass(AgentUtil.java:301)
	at dev.langchain4j.agentic.agent.AgentBuilder.<init>(AgentBuilder.java:112)
	at dev.langchain4j.agentic.AgenticServices.agentBuilder(AgenticServices.java:120)
	at dev.langchain4j.agentic.AgenticServices.createAgenticSystem(AgenticServices.java:327)
	at io.quarkiverse.langchain4j.agentic.runtime.AgenticRecorder$1.apply(AgenticRecorder.java:78)
	at io.quarkiverse.langchain4j.agentic.runtime.AgenticRecorder$1.apply(AgenticRecorder.java:62)
	at org.acme.BinanceAgenticA2aClientAgent_EtsuQd6QuW8aZwTzxOVk-1beTR0_Synthetic_Bean.createSynthetic(Unknown Source)
	... 28 more
```
if i send the same request through using the A2A java SDK, all works perfect

```
userMessage = A2A.createUserTextMessage(mapper.writeValueAsString(mapper.readValue(question, DeepChatMessages.class)), null, null);
        aiMessage = binanceA2aClientAgent.callAgent(userMessage, agentCard);
```

--- a/langchain4j-agentic-a2a/src/test/java/dev/langchain4j/agentic/a2a/A2AAgentIT.java
+++ b/langchain4j-agentic-a2a/src/test/java/dev/langchain4j/agentic/a2a/A2AAgentIT.java
@@ -7,6 +7,7 @@ import dev.langchain4j.agentic.Agent;
 import dev.langchain4j.agentic.AgenticServices;
 import dev.langchain4j.agentic.UntypedAgent;
 import dev.langchain4j.agentic.a2a.Agents.CreativeWriter;
+import dev.langchain4j.agentic.a2a.Agents.DeclarativeA2ACreativeWriter;
 import dev.langchain4j.agentic.a2a.Agents.StoryCreatorWithReview;
 import dev.langchain4j.agentic.a2a.Agents.StyleEditor;
 import dev.langchain4j.agentic.a2a.Agents.StyleReviewLoop;
@@ -154,6 +155,16 @@ public class A2AAgentIT {
         assertThat(agenticScope.readState("style")).isEqualTo("comedy");
         assertThat(story).isEqualTo(agenticScope.readState("story"));
         assertThat(agenticScope.readState("score", 0.0)).isGreaterThanOrEqualTo(0.8);
+    }
+
+    @Test
+    @Disabled("Requires A2A server to be running")
+    void declarative_single_a2a_agent_tests() {
+        DeclarativeA2ACreativeWriter writer =
+                AgenticServices.createAgenticSystem(DeclarativeA2ACreativeWriter.class, baseModel());
+
+        String story = writer.generateStory("dragons and wizards");
+        assertThat(story).isNotBlank();
     }
 
     public interface A2AStyleScorer {

--- a/langchain4j-agentic/src/main/java/dev/langchain4j/agentic/AgenticServices.java
+++ b/langchain4j-agentic/src/main/java/dev/langchain4j/agentic/AgenticServices.java
@@ -333,6 +333,11 @@ public class AgenticServices {
         T agent = createComposedAgent(agentServiceClass, chatModel, agentConfigurator);
 
         if (agent == null) {
+            Optional<Method> a2aClientMethod = getAnnotatedMethodOnClass(agentServiceClass, A2AClientAgent.class);
+            if (a2aClientMethod.isPresent()) {
+                return createA2AClient(agentServiceClass, a2aClientMethod.get());
+            }
+
             var agentBuilder = AgentBuilder.withoutDeclarativeConfiguration(agentServiceClass);
             configureAgent(agentServiceClass, chatModel, agentBuilder, agentConfigurator);
             agent = agentBuilder.build();
@@ -669,6 +674,10 @@ public class AgenticServices {
     }
 
     private static AgentExecutor createA2AClientAgent(Class<?> agentServiceClass, Method a2aMethod) {
+        return agentToExecutor(createA2AClient(agentServiceClass, a2aMethod));
+    }
+
+    private static <T> T createA2AClient(Class<T> agentServiceClass, Method a2aMethod) {
         var a2aClient = a2aMethod.getAnnotation(A2AClientAgent.class);
         var a2aClientBuilder = a2aBuilder(a2aClient.a2aServerUrl(), agentServiceClass)
                 .inputKeys(Stream.of(a2aMethod.getParameters())
@@ -683,7 +692,7 @@ public class AgenticServices {
                     a2aClientBuilder.listener(invokeStatic(method));
                 });
 
-        return agentToExecutor(a2aClientBuilder.build());
+        return a2aClientBuilder.build();
     }
 
     private static AgentExecutor createMcpClientAgent(Class<?> agentServiceClass, Method mcpMethod) {

